radix-engine/src/blueprints/pool/two_resource_pool/blueprint.rs

```diff
@@ -436,20 +436,17 @@ impl TwoResourcePoolBlueprint {
                 (false, false, false) => Ok((
```

Comment:
(I'm writing out some of my reasoning as I go through the review, apologies if the comments are somewhat lengthy).

I will be referring to `Decimal::Max` as $`\textrm{Max}_{\textrm{Decimal}}`$. I am operating with the assumption that the amount of resources in fungible buckets has a theoretical upper bound of $\textrm{Max}_{\textrm{Decimal}}$ with the understanding that the theoretical and practical limits are different. 

Currently, this bit of code is doing the following:

```math
mint\_amount = \sqrt{ contribution_1 \times contribution_2 }
```

The expression inside the square root has an upper bound of:

```math
contribution_1 \times contribution_2 = \textrm{Max}_{\textrm{Decimal}} \times \textrm{Max}_{\textrm{Decimal}} = \textrm{Max}_{\textrm{Decimal}}^2
```

Meaning that it can indeed overflow. We can refactor this into something that's guaranteed to not overflow if we express it as:

```math
mint\_amount = \sqrt{ contribution_1 \times contribution_2 } = \sqrt{contribution_1} \times \sqrt{contribution_2}
```

Expressing it in the way seen on the RHS of the equation above is safer for two reasons:

* It's impossible to overflow in this case and an unwrap is safe to do (we can choose not to unwrap and return an error instead, that's fine)
* Contributing anything less than or equal to $\textrm{Max}_{\textrm{Decimal}}$ works with no issues.
* If loss of precision is a concern then we can do all of the math with `PreciseDecimal` and then convert back to `Decimal` at the end.